src/Context/GlobalContext/StateReducer.js

```diff
@@ -47,6 +50,35 @@ export const stateReducerFun = (state, action) => {
                     categoryName: [...state.filters.categoryName, action.payload],
                 },
             };
+
+            case STAR_RATING: 
+            
+            if(action.payload === 4){
+                return{...state, filters: {...state.filters, rating: action.payload}}
```

Comment:
We can pass payload to rating without writing 4 if else. 
if( payload !== "" ) {
return rating: payload 
} 
return rating:null

Reply:
I updated this.